packages/webdriverio/src/find-strategy.js

```diff
@@ -0,0 +1,183 @@
+    if (selector.indexOf('ios=') === 0) {
+        return '-ios uiautomation'
+    }
+    // recursive element search using accessibility id
+    if (selector.indexOf('~') === 0) {
+        return 'accessibility id'
+    }
+    // class name mobile selector
+    // for iOS = UIA...
+    // for Android = android.widget
+    if (NAME_MOBILE_SELECTORS_START.some(option => selector.toLowerCase().indexOf(option) === 0)) {
+        return 'class name'
+    }
+    // use tag name strategy if selector contains a tag
+    // e.g. "<div>" or "<div />"
+    if (selector.search(/<[a-zA-Z-]+( \/)*>/g) >= 0) {
+        return 'tag name'
+    }
+    // use name strategy if selector queries elements with name attributes for JSONWP
+    // or if isMobile is used even when w3c is used
+    // e.g. "[name='myName']" or '[name="myName"]'
+    if (selector.search(/^\[name=("|')([a-zA-z0-9\-_. ]+)("|')]$/) >= 0) {
+        return 'name'
+    }
+    // allow to move up to the parent or select current element
+    if (selector === '..' || selector === '.') {
+        return 'xpath'
+    }
+    // any element with given class, id, or attribute and content
+    // e.g. h1.header=Welcome or [data-name=table-row]=Item or #content*=Intro
+    if (selector.match(new RegExp(XPATH_SELECTOR_REGEXP.map(rx => rx.source).join('')))) {
+        return 'xpath extended'
+    }
+}
+export const findStrategy = function (selector, isW3C, isMobile) {
+    /**
+     * set default strategy
+     */
+    let using = DEFAULT_STRATEGY
+    let value = selector
+
+    switch (defineStrategy(selector)) {
+    // user has specified locator strategy directly
+    case 'directly': {
+        const match = selector.match(DIRECT_SELECTOR_REGEXP)
+        if (!isMobile && isW3C && !W3C_SELECTOR_STRATEGIES.includes(match[1])) {
+            throw new Error('InvalidSelectorStrategy') // ToDo: move error to wdio-error package
+        }
+        using = match[1]
+        value = match[2]
+        break
+    }
+    case 'xpath': {
+        using = 'xpath'
+        break
+    }
+    case 'link text': {
+        using = 'link text'
+        value = selector.slice(1)
+        break
+    }
+    case 'partial link text': {
+        using = 'partial link text'
+        value = selector.slice(2)
+        break
+    }
+    case '-android uiautomator': {
+        using = '-android uiautomator'
+        value = selector.slice(8)
+        break
+    }
+    case '-ios uiautomation': {
+        using = '-ios uiautomation'
+        value = selector.slice(4)
+        break
+    }
+    case 'accessibility id': {
+        using = 'accessibility id'
+        value = selector.slice(1)
+        break
+    }
+    case 'class name': {
+        using = 'class name'
+        break
+    }
+    case 'tag name': {
+        using = 'tag name'
+        value = selector.replace(/<|>|\/|\s/g, '')
+        break
+    }
+    case 'name': {
+        if (isMobile || !isW3C) {
```

Comment:
What happens if this is not mobile but w3c?

Reply:
it will use `css selector` strategy:
```js
 Received: "css selector"

      40 |     it('should find an element using "name" method by "name" strategy if isMobile is used even when w3c is used', () => {
      41 |         const element = findStrategy('[name="searchinput"]', true, false)
    > 42 |         expect(element.using).toBe('name')
         |                               ^
      43 |         expect(element.value).toBe('searchinput')
      44 |     })

```

Reply:
I added such test case:
```js
    it('should find an element using "css selector" strategy if isMobile is false and w3c is used', () => {
        const element = findStrategy('[name="searchinput"]', true, false)
        expect(element.using).toBe('css selector')
        expect(element.value).toBe('[name="searchinput"]')
    })
```